### DISCUSSION TITLE
CodeHike and Next.js 13

### DISCUSSION THREAD

**Discussion by kai687** [code-hike/codehike#408] [Q&A] [answered]
Hi :wave:,

The docs don't include any specific info about Next.js 13 and the app router.
When I try to follow the instructions for CodeHike + Next.js and contentlayer,
I get an error: 

```
error: createContext only works in Client Components. Add the "use client" directive at the top of the file to use it.
```

Can I make this work?

**Comment by mrboen94** [comment]
You can make this work by placing "use client" at the top of the file you are using createContext in. I had the same problem, but the error message does indeed describe the solution.

```js
"use client"
import React from "react";
// etc ...
```

**Reply by kai687** [reply]
Sorry, I don't understand where I should add `use client`.

I tried to follow [the docs](https://codehike.org/docs/installation/nextjs#nextjs--code-hike) for Next.js and CodeHike, just with Next's app router instead. Since I didn't create any file with `createContext` in it, I don't know where to add 'use client'. That's why I didn't find this error message helpful and looked for more information here.

**Reply by mrboen94** [reply, marked as answer]
You usually need to use it if you have an `onClick` or similar in your component. For debugging you should be able to set it as far up the tree as possible, then move down through the files. It probably comes from one of the click events on the CH components.

Regardless this is probably a chat you'd want to take with the Next team or the NextJs discord: https://discord.com/invite/nextjs as it is not necessarily a problem with code-hike, but just a quirk of using NextJs.

For reference I have used it in this repo for a test project I did: https://github.com/mrboen94/Haiku/blob/5e91798816508a9cb393c0f7b7de706f1996c5f2/haiku/app/page.tsx#L1

However that project does not use code-hike, but it should still work by just settings `"use client"` in one of your parent files.
